docs/releases/3.0.txt

```diff
@@ -410,6 +410,10 @@ backends.
   argument, which is an iterable of fields to be returned after insert. Usually
   this is only the auto-generated primary key.
 
+* :class:`~django.db.models.AutoField`, :class:`~django.db.models.BigAutoField`, and
+  :class:`~django.db.models.SmallAutoField` are now defined as unsigned integers on
+  MySQL and SQLite. Historically, they were defined as signed integers.
```

Comment:
I would clarify that in PostgreSQL and Oracle they're already unsigned.

Reply:
No, this isn't true. In PostgreSQL the sequence will start counting from 1 as Django's auto fields are backed by serial types, but you can happily store negative values manually as the underlying columns are the integer types.

Reply:
Further, if we're going to have unsigned auto field support for other backends, we should add check constraints as we do for `PositiveIntegerField` in PostgreSQL.

Reply:
> In PostgreSQL the sequence will start counting from 1 as Django's auto fields are backed by serial types, but you can happily store negative values manually as the underlying columns are the integer types.

Is this true for identity columns as well?

Reply:
It depends on how they were defined:

```sql
$ psql postgres postgres
psql (11.5)
Type "help" for help.

postgres=# create temporary table t1 (id serial primary key);
CREATE TABLE
postgres=# insert into t1 default values;
INSERT 0 1
postgres=# insert into t1 default values;
INSERT 0 1
postgres=# insert into t1 (id) values (-1);
INSERT 0 1
postgres=# select * from t1;
 id 
────
  1
  2
 -1
(3 rows)

postgres=# create temporary table t2 (id integer primary key generated by default as identity);
CREATE TABLE
postgres=# insert into t2 default values;
INSERT 0 1
postgres=# insert into t2 default values;
INSERT 0 1
postgres=# insert into t2 (id) values (-1);
INSERT 0 1
postgres=# insert into t2 (id) overriding user value values (-1);
INSERT 0 1
postgres=# select * from t2;
 id 
────
  1
  2
 -1
  3
(4 rows)

postgres=# create temporary table t3 (id integer primary key generated always as identity);
CREATE TABLE
postgres=# insert into t3 default values;
INSERT 0 1
postgres=# insert into t3 default values;
INSERT 0 1
postgres=# insert into t3 (id) values (-1);
ERROR:  cannot insert into column "id"
DETAIL:  Column "id" is an identity column defined as GENERATED ALWAYS.
HINT:  Use OVERRIDING SYSTEM VALUE to override.
postgres=# insert into t3 (id) overriding system value values (-1);
INSERT 0 1
postgres=# select * from t3;
 id 
────
  1
  2
 -1
(3 rows)
```